--- testdemo.c	2014-05-07 16:14:58.490894701 +0800
+++ testdemo_new1.c	2014-05-07 17:15:05.122980000 +0800
@@ -4,13 +4,16 @@
 	char *name;
 	int age;
 };
-void exc_esc(){
+void exc_esc()
+{
 	int i;
-	int *p = (int *)malloc(10*sizeof(int));
-	for(i = 0; i < 10; i ++)
+	int *pt= *p = (int *)malloc(10*sizeof(int));
+	for(i= 0; i < 8; i ++)
 	  p++;
-	*p = 10;
+	*p =1;
 	printf("%d\n",*p);
+	free(pt);
+
 }
 void Noinit(){
 	int a[5],i,s = 0;
